airqo-platform/AirQo-api
# Log user activity in the Auth Service

**_WHAT DOES THIS PR DO?_**

- [x] if it is an organisation other than airqo, please do not allow the user to login. DISABLED OTHER TENANTS.
- [x] Be able to view activity logs for users.
- [x] Resolved a couple of runtime errors around the `create-user` use case.
- [x] General refactoring , with a focus on the `create-user` use case. Mostly focusing on the error responses.
- [x] Add new fields (`firstName` and `lastName`) to the inquire endpoint, following recent designs. This is the GET INVOLVED use case for the website. See on the screenshots below in this PR description
- [x] Increased the default limit for users queries from 100 to 1000, using an ENV variable.
- [x] Do QA tests on the verify email endpoint
- [x] ensure that all user inquiries endpoints function as expected.

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - [AN-322]

**_HOW DO I TEST OUT THIS PR?_**


```
cd auth-service
npm install
npm run stage-mac
```

- _After _login_ endpoint to get the `JWT token`, just run any endpoint like list users and then run the new list use activity logs endpoint to see that the user activity is now logged to the database (`logs` table)_
- _As you go about this, also check out the rest of the refactored endpoints to ensure that they are working_
- _There is also a new endpoint for listing user stats. For questions like: how many activated users do we have?_ @Codebmk 

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

- [x] [list Statistics](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-04549852-20d9-455c-bcbe-2c354743eaa3)
- [x] [Access User Activity Logs](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-6b0d295f-48ae-4415-8908-d20406f37d9b)
- [x] [verify email](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-0eaa3d72-e33e-484b-854c-a9a8940db741)
- [x] [create inquiries](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-298301ab-5236-4fb2-ae0b-2d1556c0b816)
- [x] [list inquiries](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-7e800d44-4cf3-45a1-aedd-a747859e14ea)


**_ANY SCREENSHOTS?:_** 
![image](https://user-images.githubusercontent.com/1590213/226177696-6599ed6d-e37c-44d9-8633-1a5ad00435f3.png)
![Screenshot 2023-03-28 at 08 50 36](https://user-images.githubusercontent.com/1590213/228141078-a6e17734-c22c-4ec0-82d3-32f3248656b1.png)
![Screenshot 2023-03-29 at 08 24 27](https://user-images.githubusercontent.com/1590213/228436333-bee47c03-3aab-46b9-b6e9-73a377d16c3a.png)






[AN-322]: https://airqoteam.atlassian.net/browse/AN-322?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
